controllers/dnspolicy_status.go

```diff
@@ -58,13 +58,25 @@ func (r *DNSPolicyReconciler) reconcileStatus(ctx context.Context, dnsPolicy *v1
 		return ctrl.Result{}, updateErr
 	}
 
+	// policy updated in API, emit metrics based on status conditions
+	r.emitConditionMetrics(dnsPolicy)
+
 	if kuadrant.IsTargetNotFound(specErr) {
 		return ctrl.Result{Requeue: true}, nil
 	}
 
 	return ctrl.Result{}, nil
 }
 
+func (r *DNSPolicyReconciler) emitConditionMetrics(dnsPolicy *v1alpha1.DNSPolicy) {
+	readyStatus := meta.FindStatusCondition(dnsPolicy.Status.Conditions, ReadyConditionType)
+	if readyStatus == nil {
+		dnsPolicyReady.WithLabelValues(dnsPolicy.Name, dnsPolicy.Namespace, "false").Set(1)
+	} else {
+		dnsPolicyReady.WithLabelValues(dnsPolicy.Name, dnsPolicy.Namespace, string(readyStatus.Status)).Set(1)
+	}
+}
```

Comment:
hmm this one seems odd. Wont I end up with a metric as `false` 1 and `true` 1 ?

Reply:
wondering what I am using this for? Is if "true" == 0 for > x time then alert?

Reply:
If I have {name:"test", namesapce:"test"} 0 | 1 would we acheive the same thing?

Reply:
Yeah, this probably wont behave the way you are thinking.
Metrics, and each combination of label values, should be thought of as existing all the time.
What changes is the value of those metric label combinations.
Check out https://prometheus.io/docs/practices/instrumentation/#avoid-missing-metrics

If you set a falsey metric to 1 when it's false, you should also set to to 0 when it's not false.
Otherwise you might end up with odd behaviour in graphs and alerts where your DNSPolicy shows up as 'not ready' and also as 'ready'

Another way to think about it is restarting your controller should result in the same metrics and values being emitted. 
If you do actually want a metric to no longer be scraped, theres [Delete](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#MetricVec.Delete) and other functions,

Reply:
ok, so I need to emit 1 when ready and otherwise 0? That makes sense to me. Rereading the above I'm not actually clear on what I was thinking.